spring-kafka/src/main/java/org/springframework/kafka/listener/AbstractMessageListenerContainer.java

```diff
@@ -275,6 +277,14 @@ public boolean isRunning() {
 		return this.running;
 	}
 
+	protected void setFenced(boolean fenced) {
+		this.fenced = fenced;
+	}
+
+	public boolean isFenced() {
```

Comment:
I even don't see a reason in this one.
That is fully internal logic of the container architecture.
And if you have to deal with this new public API, then the logic in your project is off.
The consumer of the framework must use it in a proper way.
So, any child containers must be controlled outside of the parent container, when that parent container has abandoned them.

Would you sharing with us some idea what is the use-case that we would need to deal with child lifecycle individually, but not via its parent?